pkg/controllers/ipaddressallocation/ipaddressallocation_controller.go

```diff
@@ -42,23 +42,25 @@ type IPAddressAllocationReconciler struct {
 	Recorder   record.EventRecorder
 }
 
-func deleteSuccess(r *IPAddressAllocationReconciler, _ context.Context, o *v1alpha1.IPAddressAllocation) {
-	r.Recorder.Event(o, v1.EventTypeNormal, common.ReasonSuccessfulDelete, "IPAddressAllocation CR has been successfully deleted")
+func deleteSuccess(r *IPAddressAllocationReconciler, namespacedName types.NamespacedName) {
+	log.Info("Successfully deleted IPAddressAllocation CR", "IPAddressAllocation", namespacedName)
 	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteSuccessTotal, MetricResType)
 }
 
-func deleteFail(r *IPAddressAllocationReconciler, c context.Context, o *v1alpha1.IPAddressAllocation, e *error) {
-	r.Recorder.Event(o, v1.EventTypeWarning, common.ReasonFailDelete, fmt.Sprintf("%v", *e))
+func deleteFail(r *IPAddressAllocationReconciler, namespacedName types.NamespacedName, e *error) {
```

Comment:
Why use a pointer for the error in signature? Can we use type `error` directly?

Reply:
Updated, thanks